### DISCUSSION TITLE
How to output execution time of each test case

### DISCUSSION THREAD

**Discussion by ffbh123456** [pytest-dev/pytest#8268] [Q&A]
I write some test cases, and then I use **pytest** to run these test cases. I hope to output the execution time of each test case on the console. What should I do?

**Comment by The-Compiler** [comment]
From `pytest --help`:

> --durations=N         show N slowest setup/test durations (N=0 for all).

**Reply by ffbh123456** [reply]
Yes, this is one way, but it outputs after all test cases are executed. I hope to output the running time of this test case immediately after each test case is executed

**Comment by ffbh123456** [comment]
@The-Compiler 
I was wondering if I can add a callback function, which is called every time a test case is completed so that I can modify the output of the case by myself

**Reply by The-Compiler** [reply]
See [writing plugins](https://docs.pytest.org/en/latest/writing_plugins.html) and the [hooks reference](https://docs.pytest.org/en/latest/reference.html#hook-reference). I'm afraid I don't have time for more details at the moment, but that should hopefully get you started!

**Comment by kaddkaka** [comment]
A feature request to make -v print the execution time of each test seems reasonable, no?
